lib/erb_lint/linters/final_newline.rb

```diff
@@ -10,22 +10,22 @@ def initialize(config)
         @new_lines_should_be_present = config['present'].nil? ? true : config['present']
       end
 
-      protected
-
-      def lint_lines(lines)
+      def lint_file(file_tree)
         errors = []
-        return errors if lines.empty?
+        return errors if file_tree.children.size == 1
 
-        ends_with_newline = lines.last.chars[-1] == "\n"
+        end_marker = file_tree.search(Parser::END_MARKER_NAME).first
+        last_child = end_marker.previous_sibling
+        ends_with_newline = last_child.text? && last_child.text.chars[-1] == "\n"
 
         if @new_lines_should_be_present && !ends_with_newline
```

Comment:
Please move this `generate_errors`-like behaviour to a separate class or encapsulate it in a method.